src/allocation.rs

```diff
@@ -149,6 +149,42 @@ pub fn zeroed_slice_box<T: Zeroable>(length: usize) -> Box<[T]> {
   try_zeroed_slice_box(length).unwrap()
 }
 
+/// Allocates a `Arc<T>` with all contents being zeroed out.
+pub fn zeroed_arc<T: Zeroable>() -> Arc<T> {
```

Comment:
```suggestion
/// Allocates a `Arc<T>` with all contents being zeroed out.
#[cfg(target_has_atomic = "ptr")]
pub fn zeroed_arc<T: Zeroable>() -> Arc<T> {
```

Reply:
I also forgot `cfg(feature = "alloc_uninit")`, surprised that the CI didn't catch it